src/test/java/uk/nhs/prm/e2etests/test/repositorye2etest/RepositoryNackTests.java

```diff
@@ -169,4 +180,264 @@ void shouldSend06NackWhenIncompleteInbound(ConversationTransferStatus transferSt
 
         assertNackMessageReceived("06");
     }
+
+    @Test
+    void shouldSend10NackWhenEHRNotFound() {
+        final String nhsNumber = PATIENT_WITH_SMALL_EHR_IN_REPO_AND_MOF_SET_TO_TPP.nhsNumber();
+
+        //set up a complete inbound transfer
+        transferTrackerService.saveConversation(ConversationRecord.builder()
+                .inboundConversationId(inboundConversationId)
+                .nhsNumber(nhsNumber)
+                .transferStatus(INBOUND_COMPLETE.name())
+                .nemsMessageId(nemsMessageId)
+                .sourceGp(senderOdsCode)
+                .associatedTest(testName)
+                .build()
+        );
+
+        //do not transfer the EHR, continue to set up the outbound request
+        EhrRequestTemplateContext ehrRequestContext = EhrRequestTemplateContext.builder()
+                .outboundConversationId(outboundConversationId)
+                .nhsNumber(nhsNumber)
+                .senderOdsCode(senderOdsCode)
+                .build();
+
+        String inboundMessage = templatingService.getTemplatedString(EHR_REQUEST, ehrRequestContext);
+
+        // When the message is received via the mhsInboundQueue
+        mhsInboundQueue.sendMessage(inboundMessage, outboundConversationId);
+
+        assertNackMessageReceived("10");
+    }
+
+    @Test
+    void shouldSend10NackWhenEHRNotFoundWithInboundId(){
+        final String nhsNumber = PATIENT_WITH_SMALL_EHR_IN_REPO_AND_MOF_SET_TO_TPP.nhsNumber();
+        final String senderOdsCode = TPP_PTL_INT.odsCode();
+        final String asidCode = TPP_PTL_INT.asidCode();
+
+        log.info("nhsNumber: " + nhsNumber);
+        log.info("senderOdsCode: " + senderOdsCode);
+        log.info("asidCode: " + asidCode);
+
+        /*
+        ORC-IN
```

Comment:
It's worth checking the other tests - There should be a one-liner shorthand to ingest a small EHR into the repo, this is fine to stay if it's doing something different, but I thought I'd mention to keep the test as concise as possible